### ISSUE TITLE
decodeTagged returns bottom instead of Nothing

### ISSUE THREAD

**Issue by lspitzner** [mstksg/tagged-binary#1]
decoding with the wrong type throws an error. testcase:

``` .hs
main :: IO ()
main = do
  let decoded :: Maybe Float
      decoded = decodeTagged $ encodeTagged (42::Int)
  print decoded
```

```
Data.Binary.Get.runGet at position 23: demandInput: not enough bytes
```

The documentation seems to suggest that decodeTagged would return Nothing.
ghc-7.10.1, base-4.8.0.0, binary-0.7.3.0, tagged-binary-0.1.2.0

**Comment by mstksg** [comment]
Should be fixed for v0.2.0.0 c04b4b4b67 :) should be up on hackage now. let me know if you're still having issues

**Comment by lspitzner** [comment]
very nice, thanks!
